core/trino-parser/src/main/java/io/trino/sql/tree/GroupingSets.java

```diff
@@ -32,7 +33,8 @@ public enum Type
     {
         EXPLICIT,
         ROLLUP,
-        CUBE
+        CUBE,
+        AUTO
```

Comment:
Create a separate class to represent AUTO. It should extend `GroupingElement` and be parallel to SimpleGroupBy and GroupingSets

Reply:
Added AutoGroupBy class extending GroupingElement.